core/jobs/batch_jobs/state_validation_jobs.py

```diff
@@ -0,0 +1,133 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for exploration state"""
+
+from __future__ import annotations
+
+from core.domain import exp_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import bs4
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import exp_models
+
+(exp_models, ) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class GetNumberOfExpStatesHavingEmptyImageFieldJob(base_jobs.JobBase):
+    """Job that returns exploration id and exploration states that have
+    filepath-with-value field as an empty string.
+    """
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        exploration_with_invalid_states = (
+            total_explorations
+            | 'Combine exploration id and states' >> beam.Map(
+                lambda exp: (exp.id, exp.states))
+            | 'Get only states' >> beam.Map(
+                lambda objects: (objects[0], self.get_states(objects[1]))
+            )
+            | 'Get invalid states' >> beam.Filter(
+                lambda objects: self.check_invalid(objects[1])
+            )
+            | 'Remove empty string' >> beam.Map(
+                lambda objects: (objects[0], self.extract_name(objects[1]))
+            )
+        )
+
+        report_number_of_invalid_states_queried = (
+            exploration_with_invalid_states
+            | 'Report count of exp models' >> (
+                job_result_transforms.CountObjectsToJobRunResult('STATES'))
+        )
+
+        report_invalid_states = (
+            exploration_with_invalid_states
+            | 'Save info on invalid exps' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                    'The id of exp is %s and the erroneous states are %s'
+                    % (objects[0], objects[1])
+                ))
+        )
+
+        return (
+            (
+                report_invalid_states,
+                report_number_of_invalid_states_queried
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+    def get_states(self, states_dict):
```

Comment:
Missing types.

Reply:
Ditto below.

Reply:
@vojtechjelinek 
just to be sure, here types mean the type of the return in the function, right. 
In this case, it will be something like `-> list`
Thanks

Reply:
Yes

Reply:
Please add more concrete types (like `List[str]` etc.), also add types for arguments of the functions.

Reply:
Done, Thanks

Reply:
@vojtechjelinek I think I have made some mistakes because the type checks are failing, The errors are
```
scripts/run_frontend_tests.py:125: error: unused 'type: ignore' comment
scripts/run_frontend_tests.py:129: error: unused 'type: ignore' comment
scripts/run_frontend_tests.py:170: error: unused 'type: ignore' comment
```
Not sure how to fix this.
Thanks

Reply:
This should now be fixed just merge from develop